lib/FindLocation/FindLocationLookup.js

```diff
@@ -0,0 +1,178 @@
+import keyBy from 'lodash/keyBy';
+import PropTypes from 'prop-types';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import { ColumnManager } from '@folio/stripes/smart-components';
+
+import { EVENT_EMMITER_EVENTS } from '../constants';
+import { FindRecords } from '../FindRecords';
+import {
+  useCampuses,
+  useEventEmitter,
+  useInstitutions,
+  useLibraries,
+} from '../hooks';
+import {
+  COLUMN_MAPPING,
+  COLUMN_NAMES,
+  NON_TOGGLEABLE_COLUMNS,
+  SORTABLE_COLUMNS,
+} from './configs';
+import { FindLocationFilters } from './FindLocationFilters';
+import { useLocationsRecords } from './useLocationsRecords';
+
+export const FindLocationLookup = (props) => {
+  const {
+    filterRecords,
+    idPrefix,
+    initialSelected,
+    isMultiSelect,
+    modalLabel,
+    onRecordsSelect,
+    resultsPaneTitle,
+    sortableColumns,
+    ...rest
+  } = props;
+
+  const intl = useIntl();
+  const eventEmitter = useEventEmitter();
+
+  const [selectedLocations, setSelectedLocations] = useState({});
+  const [filters, setFilters] = useState({});
+
+  const { institutions, isLoading: isInstitutionsLoading } = useInstitutions();
+  const { campuses, isLoading: isCampusesLoading } = useCampuses();
+  const { libraries, isLoading: isLibrariesLoading } = useLibraries();
+
+  const institutionsMap = useMemo(() => keyBy(institutions, 'id'), [institutions]);
+  const campusesMap = useMemo(() => keyBy(campuses, 'id'), [campuses]);
+  const librariesMap = useMemo(() => keyBy(libraries, 'id'), [libraries]);
+
+  const { locations, isLoading: isLocationsLoading } = useLocationsRecords({
+    filters,
+    filterRecords,
+    selectedLocations,
+    initialSelected,
+    institutionsMap,
+    campusesMap,
+    librariesMap,
+  });
+
+  /*
+    Subscribe on changes in selected records.
+  */
+  useEffect(() => {
+    const eventType = EVENT_EMMITER_EVENTS.FIND_RECORDS_SELECTED_CHANGED;
+    const callback = ({ detail: assignment }) => setSelectedLocations(assignment);
+
+    eventEmitter.on(eventType, callback);
+
+    return () => {
+      eventEmitter.off(eventType, callback);
+    };
+  }, [eventEmitter]);
+
+  const columnMapping = useMemo(() => ({
+    ...COLUMN_MAPPING,
+    ...(
+      isMultiSelect
+        ? { [COLUMN_NAMES.isAssigned]: <FormattedMessage id="stripes-acq-components.find-location.results.column.assignment" /> }
+        : {}
+    ),
+  }), [isMultiSelect]);
+
+  const resultsFormatter = useMemo(() => ({
```

Comment:
can we move formatter in some utils/helpers ?

Reply:
Yes, makes sense. I created a separate hook `useLocationsList` to prepare list-related props like formatted and column mapping, thanks!